sha1/tests/mod.rs

```diff
@@ -13,3 +13,89 @@ fn sha1_rand() {
         hex!("7e565a25a8b123e9881addbcedcd927b23377a78"),
     );
 }
+
+#[cfg(all(feature = "collision", not(feature = "force-soft")))]
+#[test]
+fn shambles_1() {
+    collision_test(
+        &include_bytes!("./data/sha-mbles-1.bin")[..],
+        hex!("8ac60ba76f1999a1ab70223f225aefdc78d4ddc0"),
+        hex!("4f3d9be4a472c4dae83c6314aa6c36a064c1fd14"),
+    )
+}
+
+#[cfg(all(feature = "collision", not(feature = "force-soft")))]
+#[test]
+fn shambles_2() {
+    collision_test(
+        &include_bytes!("./data/sha-mbles-2.bin")[..],
+        hex!("8ac60ba76f1999a1ab70223f225aefdc78d4ddc0"),
+        hex!("9ed5d77a4f48be1dbf3e9e15650733eb850897f2"),
+    )
+}
+
+#[cfg(all(feature = "collision", not(feature = "force-soft")))]
+#[test]
+fn shattered_1() {
+    collision_test(
+        &include_bytes!("./data/shattered-1.pdf")[..],
+        hex!("38762cf7f55934b34d179ae6a4c80cadccbb7f0a"),
+        hex!("16e96b70000dd1e7c85b8368ee197754400e58ec"),
+    )
+}
+
+#[cfg(all(feature = "collision", not(feature = "force-soft")))]
+#[test]
+fn shattered_2() {
+    collision_test(
+        &include_bytes!("./data/shattered-2.pdf")[..],
+        hex!("38762cf7f55934b34d179ae6a4c80cadccbb7f0a"),
+        hex!("e1761773e6a35916d99f891b77663e6405313587"),
+    )
+}
+
```

Comment:
I think an explicit test for the default values of `checked::Config`, somewhere, would have helped me in understanding the `collision_test!` macro, as it wasn't obvious to me that many of these flags that are turned off, are on by default.

Reply:
update the api to use a builder pattern, which should make it clearer

Reply:
Unfortunately I am spoiled already, knowing what I know. However, since it's the 'checked' version of the hash, I think it's absolutely fair to assume that all checking features are enabled by default unless they are disabled explicitly. The builder pattern doesn't make the more discoverable in tests I think (but an assertion for the desired value of the fields would have made it clear).

Using a builder for that might be a matter of whether that pattern was used in this crate before or not. I'd personally lean towards fields in a struct mainly because they are well-known, won't change (it's all part of the spec implemented here), and are easy enough to use.